src/compiler/propagation/type_propagator.cc

```diff
@@ -1173,6 +1173,13 @@ static TypeScope* process(TypeScope* scope, uint8* bcp, std::vector<Worklist*>&
     }
   OPCODE_END();
 
+  OPCODE_BEGIN(BRANCH_IF_NOT_NULL);
```

Comment:
Maybe this should also be treated as an unconditional branch if the variable cannot be null?